config/clusters/2i2c/enc-cilogon-clients.secret.yaml

```diff
@@ -0,0 +1,16 @@
+2i2c-dask-staging:
```

Comment:
This seems confusing to me. Should this be named like that?

Reply:
This name comes from https://github.com/2i2c-org/infrastructure/blob/6bf2e153256fd15b2793ea8229e1da2b75efe7cc/deployer/hub.py#L135.

We had this convention before too https://github.com/2i2c-org/infrastructure/blob/dd1517d2414f6f49ba22d325074bf488908dbb0e/deployer/hub.py#L131-L133

This is how the 2i2c Auth0 applications are called https://manage.auth0.com/dashboard/us/2i2c/applications.

This client name gets written to this encrypted file [here](https://github.com/2i2c-org/infrastructure/blob/6bf2e153256fd15b2793ea8229e1da2b75efe7cc/deployer/cilogon_auth.py#L160-L170). The reason we are storing these for cilogon and we didn't for Auth0 is because cilogon retrieves details about a client only if provided with the client's id and not its name (`2i2c-dask-staging` is such a name). Also, it doesn't retrieve you the client's secret afterwards, only when it gets created.

Reply:
Ah thanks for the thorough comment and references @GeorgianaElena! Sorry for the slow followup on this! I see now that the structure has changes, so I figure this may be a resolved comment.

---

I think a key confusion point for me was regarding that this was a .YAML file that we load, but it wasn't a helm chart values file and didn't seem to be a cluster.yaml like file either - so then it seemed like there wasn't a schema to validate this configuration.

I think that for any configuration it is probably good if we avoid having keys that contain data by themselves though, so instead of having a name as a key like...

```yaml
this-is-some-cluster-hub-name: some value
```

We could have something like....

```yaml
clusterHub:
  name: this-is-some-cluster-hub-name
  someProperty: some-value
```

This latter would be easy to have a schema for, and its data would be described in the YAML structure etc.

Reply:
I see https://github.com/2i2c-org/infrastructure/pull/1089#pullrequestreview-912171302 which is related, and yes - this seems to be resolved!